activerecord/test/cases/adapters/postgresql/uuid_test.rb

```diff
@@ -9,6 +9,10 @@ def connection
   def drop_table(name)
     connection.drop_table name, if_exists: true
   end
+
+  def uuid_function
+    connection.supports_pgcrypto_uuid? ? 'gen_random_uuid()' : 'uuid_generate_v4()'
```

Comment:
Wouldn't this mean that upgrading the PG database would case migrations to generate a different result? This might be unexpected at times.

Using a configuration option that's set in an initializer could do the trick. Quite a bit of work for a simple change like this though.

 @matthewd any thoughts how we could make this less intrusive?